test/realtime/failure.test.js

```diff
@@ -328,6 +358,58 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
       });
     });
 
+    utils.arrForEach(availableTransports, function (transport) {
+      it('channel_backoff_' + transport, function (done) {
+        var channelRetryTimeout = 150;
+        var realtime = helper.AblyRealtime({
+            channelRetryTimeout: channelRetryTimeout,
+            realtimeRequestTimeout: 1,
+            transports: [transport],
+          }),
+          channel = realtime.channels.get('failed_attach'),
+          originalOnMessage = channel.onMessage.bind(channel),
+          retryCount = 0;
+
+        var performance = isBrowser ? window.performance : require('perf_hooks').performance;
+
+        channel.onMessage = function (message) {
+          // Ignore ATTACHED messages
+          if (message.action === 11) {
+            return;
+          }
+          originalOnMessage(message);
+        };
+
+        channel.attach(function (err) {
+          if (err) {
+            var lastSuspended = performance.now();
+            channel.on(function (stateChange) {
+              if (stateChange.current === 'suspended') {
+                if (retryCount > 4) {
+                  closeAndFinish(done, realtime);
+                  return;
+                }
+                var elapsedSinceSuspneded = performance.now() - lastSuspended;
+                lastSuspended = performance.now();
+                try {
+                  // JS timers don't work precisely and realtimeRequestTimeout can't be 0 so add 5ms to the max timeout length each retry
+                  expect(elapsedSinceSuspneded).to.be.below(
+                    channelRetryTimeout + Math.min(retryCount, 3) * 50 + 5 * (retryCount + 1)
+                  );
+                  expect(elapsedSinceSuspneded).to.be.above(0.8 * (channelRetryTimeout + Math.min(retryCount, 3) * 50));
```

Comment:
This generalized calculated lower bound breaks for any value other than 150. 
If `channelRetryTimeout` is 50, for first retry, calculated value is `8`.
While actual value generated using `getRetryTime` is `4.7`

Reply:
resolved as a part of #1344